stablehlo/dialect/TypeInference.cpp

```diff
@@ -259,10 +259,25 @@ LogicalResult verifyQPerAxisScaleAndZeroPointConstraints(
 //===----------------------------------------------------------------------===//
 
 // Checks if the vector `nums` has duplicates.
-bool isUnique(ArrayRef<int64_t> nums) {
+const auto isUnique = [](const ArrayRef<int64_t> nums) {
   llvm::SmallDenseSet<int64_t> set(nums.begin(), nums.end());
   return set.size() == nums.size();
-}
+};
+
+const auto checkDimsDistinct =
```

Comment:
After some time of reviewing, I realized that you just refactored this function out. This requires a few changes, but I think this is worth the effort since it makes the code more concise. Lmk if this is too much, I can make a follow up PR if needed.

Reply:
llvm::concat<const int64_t> doesn't return an ArrayRef, so this would require isUnique to accept a templated container, is that what you were thinking?

Reply:
You're right, then let's leave the current changes, and I'll revisit this later.